drivers/ethernet/eth_lowRISC.c

```diff
@@ -0,0 +1,357 @@
+
+    // Check if there is Rx Data available ...
+    if (rsr & RSR_RECV_DONE_MASK)
+    {
+        // Read the rx length for the buffer slot we are processing. There are
+        // up-to 8 of these. TODO: Verify the << 3.
+        const int rx_len = eth_read(priv, RPLR_OFFSET + ((first & 0x7) << 3));
+        const int rlen = (rx_len & RPLR_LENGTH_MASK) - 4;  // Discard FCS bytes.
+        return rlen;
+    }
+    return 0;
+}
+
+static void inline eth_copyin(struct net_local_lr *priv, u8 *data, int len,
+                              int start)
+{
+    int i, rnd = ((len - 1) | 7) + 1;
+    volatile u64 *eth_base = (volatile u64 *)(priv->ioaddr);
+    if (!(((size_t)data) & 7))
+    {
+        u64 *ptr = (u64 *)data;
+        for (i = 0; i < rnd / 8; i++) ptr[i] = eth_base[start + i];
+    }
+    else  // We can't unfortunately rely on the skb being word aligned
+    {
+        for (i = 0; i < rnd / 8; i++)
+        {
+            u64 notptr = eth_base[start + i];
+            memcpy(data + (i << 3), &notptr, sizeof(u64));
+        }
+    }
+}
+
+/*
+ *  Fetch `len` number of bytes from the current `first` buffer.
+ *
+ *  First, Last == pointers to indicate full or empty
+ *  Next  == buffer slot to read data into for the device (only increments)
+ *  Full (buffer slot)
+ */
+int lr_eth_recv(struct net_local_lr *priv, u8 *buf, const int len)
+{
+    assert(len > 0);
+
+    const int rsr = eth_read(priv, RSR_OFFSET);
+    const int first = rsr & RSR_RECV_FIRST_MASK;
+//    const int next = rsr & RSR_RECV_NEXT_MASK >> 4;
+
+    const int start = RXBUFF_OFFSET / 8 + ((first & 7) << 8);
+    eth_copyin(priv, buf, len, start);
+
+    /*
+     *  Advance first buffer by 1 slot (that we consumed above).
+     */
+    eth_write(priv, LR_WR_FIRST_BUFFER_PTR, (first + 1) & 0xF);
+    return len;
+}
+
+static struct net_if *get_iface(struct net_local_lr *ctx, uint16_t vlan_tag)
+{
+#if defined(CONFIG_NET_VLAN)
+	struct net_if *iface;
+
+	iface = net_eth_get_vlan_iface(ctx->iface, vlan_tag);
+	if (!iface) {
+		return ctx->iface;
+	}
+
+	return iface;
+#else
+	ARG_UNUSED(vlan_tag);
+
+	return ctx->iface;
+#endif
+}
+
+void lr_eth_enable_irq(struct net_local_lr *priv) { eth_enable_irq(priv); }
+void lr_eth_disable_irq(struct net_local_lr *priv) { eth_disable_irq(priv); }
+
+/*
+ *  Trigger an IRQ - this is done on driver startup and if the rx buffer is
+ *  empty and we did not fetch any packets.
+ */
+void lr_eth_trigger_irq(struct net_local_lr *priv)
+{
+    int rsr = eth_read(priv, RSR_OFFSET);
+    const int first = rsr & RSR_RECV_FIRST_MASK;
+    eth_write(priv, LR_WR_FIRST_BUFFER_PTR, (first + 1) & 0xF);
+}
+
+static void lr_isr(struct device *dev)
+{
+   ssize_t n; // Bytes rx'd
+   struct net_local_lr *priv = dev->data;
+   struct net_pkt *pkt = NULL;
+   uint16_t vlan_tag = NET_VLAN_TAG_UNSPEC;
+
+   if ((n = lr_eth_recv_size(priv)) > 0)
+   {
+
+		pkt = net_pkt_rx_alloc_with_buffer(priv->iface, n, AF_UNSPEC, 0,
```

Comment:
I suspect somewhere we are mixing tabs and spaces. The recogni and I suspect zephyr standard is to use spaces instead of tabs and set tabs to 4 spaces.